### PR TITLE
SAK-32252 Update to central mockito version.

### PR DESCRIPTION
**Pull request by buckett**
This defines mockito centrally and has all projects use that version, this reduces the number of versions of mockito we need to download.

### CODE REVIEW

**Comment by buckett** [comment]
I've left in the `<scope>test</scope>` so it's easier to see that a dependency is a for test although strictly isn't not required.

**Comment by ern** [comment]
I wonder since we include junit in every project, I wouldn't be against doing the same for mockito as its widely used and its only for scope test?

Otherwise yeah this looks good.

**Comment by buckett** [comment]
I'd prefer to keep it explicit as to what needs Mockito as it does also pull in other dependencies,

File: search/elasticsearch/impl/src/test/org/sakaiproject/search/elasticsearch/ElasticSearchTest.java [hunk 1]
```diff
@@ -134,8 +130,6 @@ public void createTestResources() {
             resources.put(name, resource1);
             events.add(newEvent);
             when(newEvent.getResource()).thenReturn(resource1.getName());
-            when(newEvent.getContext()).thenReturn(siteId);
```
Comment (buckett) [inline search/elasticsearch/impl/src/test/org/sakaiproject/search/elasticsearch/ElasticSearchTest.java]: Yep, and newer mockito fails tests if they aren't used.

Comment (buckett) [inline search/elasticsearch/impl/src/test/org/sakaiproject/search/elasticsearch/ElasticSearchTest.java]: You can configure the old behaviour but I think it's cleaner to have only the required mocks

File: kernel/api/pom.xml [hunk 2]
```diff
@@ -19,7 +19,7 @@
   <properties>
     <deploy.target>shared</deploy.target>
     <kernel.basedir>${basedir}/..</kernel.basedir>
-    <powermock.version>1.6.2</powermock.version>
```
Comment (jonespm) [inline kernel/api/pom.xml]: Fixed in #4009

File: master/pom.xml [hunk 3]
```diff
@@ -1651,6 +1652,18 @@
         <scope>test</scope>
       </dependency>
       <dependency>
+        <groupId>org.mockito</groupId>
+        <artifactId>mockito-core</artifactId>
+        <version>${sakai.mockito.version}</version>
+        <scope>test</scope>
+      </dependency>
+      <dependency>
```
Comment (jonespm) [inline master/pom.xml]: Fixed in #4009

**Pull request merged**